docs/assertions/snowflake/snowflake_dmfs.md

```diff
@@ -0,0 +1,167 @@
+
+By default in a folder called `target`. You can use config option `-o <output_folder>` in `compile` command to write these compile artifacts in another folder.
+
+Each of these artifacts will be important for the next steps in the process.
+
+_dmf_definitions.sql_
+
+This file stores the SQL code for the DMFs that will be registered in Snowflake, generated
+from your YAML assertion definitions during the compile step.
+
+```sql
+# Example dmf_definitions.sql
+
+-- Start of Assertion 5c32eef47bd763fece7d21c7cbf6c659
+
+            CREATE or REPLACE DATA METRIC FUNCTION
+            test_db.datahub_dmfs.datahub__5c32eef47bd763fece7d21c7cbf6c659 (ARGT TABLE(col_date DATE))
+            RETURNS NUMBER
+            COMMENT = 'Created via DataHub for assertion urn:li:assertion:5c32eef47bd763fece7d21c7cbf6c659 of type volume'
+            AS
+            $$
+            select case when metric <= 1000 then 1 else 0 end from (select count(*) as metric from TEST_DB.PUBLIC.TEST_ASSERTIONS_ALL_TIMES )
+            $$;
+
+-- End of Assertion 5c32eef47bd763fece7d21c7cbf6c659
+....
+```
+
+_dmf_associations.sql_
+
+This file stores the SQL code for associating with the target table,
+along with scheduling the generated DMFs to run on at particular times.
+
+```sql
+# Example dmf_associations.sql
+
+-- Start of Assertion 5c32eef47bd763fece7d21c7cbf6c659
+
+            ALTER TABLE TEST_DB.PUBLIC.TEST_ASSERTIONS_ALL_TIMES SET DATA_METRIC_SCHEDULE = 'TRIGGER_ON_CHANGES';
+            ALTER TABLE TEST_DB.PUBLIC.TEST_ASSERTIONS_ALL_TIMES ADD DATA METRIC FUNCTION test_db.datahub_dmfs.datahub__5c32eef47bd763fece7d21c7cbf6c659 ON (col_date);
+
+-- End of Assertion 5c32eef47bd763fece7d21c7cbf6c659
+....
+```
+
+
+#### Step 4. Register the compiled DMFs in your Snowflake environment
+
+Next, you'll need to run the generated SQL from the files output in Step 3 in Snowflake.
+
+You can achieve this either by running the SQL files directly in the Snowflake UI, or by using the SnowSQL CLI tool:
+
+```bash
+snowsql -f dmf_definitions.sql
+snowsql -f dmf_associations.sql
+```
+
+
+#### Step 5. Run ingestion to report the results back into DataHub
+
+Once you've registered the DMFs, they will be automatically executed, either when the target table is updated or on a fixed
+schedule.
+
+To report the results of the generated Data Quality assertions back into DataHub, you'll need to run the DataHub ingestion process with a special configuration
+flag: `include_assertion_results: true`:
+
+```yaml
+# Your DataHub Snowflake Recipe
+source:
+  type: snowflake
+  config:
+    # ...
+    include_assertion_results: True
+    # ...
+```
+
+During ingestion we will query for the latest DMF results stored in Snowflake, convert them into DataHub Assertion Results, and report the results back into DataHub during your ingestion process
+either via CLI or the UI visible as normal assertions. 
+
+`datahub ingest -c snowflake.yml`
+
```

Comment:
I'm not sure that makes sense. If we redirect to Snowflake DMF documentation it's sufficiet, otherwise we'll have to constantly update this file as things change on the snowflake side.

Reply:
I'll look for opportunities to incorporate.